cms/static/js/views/uploads.js

```diff
@@ -26,11 +26,43 @@ define(["jquery", "underscore", "gettext", "js/views/modals/base_modal", "jquery
                 BaseModal.prototype.addActionButtons.call(this);
             },
 
+            loadImageFile: function (file, callback) {
+                // Load an image file.
+                var dimensions = this.model.get('dimensions'),
+                    image = new Image();
+                image.onload = function() { callback(image) };
+                image.src = URL.createObjectURL(file);
+            },
+
             renderContents: function() {
-                var isValid = this.model.isValid(),
-                    selectedFile = this.model.get('selectedFile'),
-                    oldInput = this.$("input[type=file]").get(0);
-                BaseModal.prototype.renderContents.call(this);
+                var selectedFile = this.model.get('selectedFile');
+                var mimeTypes = this.model.get('mimeTypes');
+                var self = this;
+
+                function isSelectedFileAnImage() {
+                    //Check: Is selected file an image object and have dimensions.
+                    return (selectedFile &&
+                        $.inArray(selectedFile.type, ["image/gif", "image/jpeg", "image/png"]) > -1 &&
+                        !$.isEmptyObject(self.model.get('imageDimensions')))
+                }
+
+                if (isSelectedFileAnImage()) {
+                    //load the selected image file so its dimensions are accessible to measure.
+                    this.loadImageFile(selectedFile, function(uploadedImage) {
```

Comment:
Here we can put an `if` statement to check if their is a restriction on dimension of image file and the checking dimensions instead of doing it in `isSelectedFileAnImage` function.